File: task-03/Elixir/3/Sub-Task3.exs
defmodule Diamond do
  def main do
    IO.write("Enter a number: ")
    n = IO.gets("") |> String.trim() |> String.to_integer()
    print_diamond(n)
  end

  defp print_diamond(n) do
    for i <- 0..(n-1) do
      IO.write(String.duplicate(" ", n - i - 1))
      IO.write(String.duplicate("*", 2 * i + 1))
      IO.puts("")
    end

    for i <- (n-2)..0 do
      IO.write(String.duplicate(" ", n - i - 1))
      IO.write(String.duplicate("*", 2 * i + 1))
      IO.puts("")
    end
  end
end

Diamond.main()

File: task-03/Ruby/3/Sub-Task3.rb
def print_diamond(n)
    (0...n).each do |i|
      print ' ' * (n - i - 1)
      print '*' * (2 * i + 1)
      puts
    end
  
    (n - 2).downto(0) do |i|
      print ' ' * (n - i - 1)
      print '*' * (2 * i + 1)
      puts
    end
  end
  
  print 'Enter a number: '
  n = gets.to_i
  print_diamond(n)
  